companies/serializers.py

```diff
@@ -117,3 +111,15 @@ def create(self, validated_data: dict) -> CompanyProfile:
 
         return company
 
+
+class CompanyFollowersSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CompanyFollowers
+        fields = ["startup", "created_at"]
+
+    def validate(self, data):
+        """Ensure that company has the right type."""
+        startup = data.get("startup")
+        if startup.type != CompanyType.STARTUP:
+            raise serializers.ValidationError({"startup": "The company must be a startup."})
+        return data
```

Comment:
The model `CompanyFollowers` requires both an `investor` and a `startup`. You should check if the investor is an `enterprise` before saving.